SU-SWS/stanford_paragraph_types
# Hsdo 751 defaults

# READY FOR REVIEW

# Summary
- Allows multiple default options for paragraph fields. Took the idea from [Paragraphs Defaults](https://www.drupal.org/project/paragraphs_defaults) but allowed multiple. Also allows for export/import via features.
- After creating default configurations, there are two ways to load them on the form.
  1. Via the button that is added to the form in the field widget.
  2. Via GET parameter 'layout' ex: node/add/stanford-story-page?layout=field_paragraphs_field-layout_machine_name

# Needed By (Date)
- Unknown

# Steps to Test
1. Checkout the branch, enable stanford_paragraph_types & stanford_paragraphs_defaults
2. Add a paragraph field to an entity such as a node. 
3. View admin/structure/paragraphs/stanford-defaults - you should see a table to add a new default configuration for you paragraph field
4. Add one or more new default settings.
5. Go to add a new entity that the field is on.
6. Your default buttons should appear in the field. Click on one of the buttons to load that layout.
7. Also to load the layout add the parameter `layout=field_my_field-layout_machine_name` to the url. This should load the layout on first load. (could be linked in a menu item.)
8. The buttons should not be visible if existing data is there like after clicking one of the buttons or editing an existing entity.
9. Create a new feature with those paragraphs defaults settings and save to the site directory (don't enable yet).
10. Delete the default configurations you added to the feature.
11. Enable the feature you created. the default settings should return and the feature should be in default state.

# Notes
Images & Files export with the feature as well. They should remain in the `assets` directory to keep the feature in a default state.
Should it use [System stream wrapper](https://www.drupal.org/project/system_stream_wrapper) to prevent the need to copy the asset files?

# Associated people
@cjwest @sherakama @jbickar
